File: Biblioteca/frontend/features/biblioteca/models/libro.model.ts
export type EstadoLectura =
  | 'Leyendo'
  | 'Leído'
  | 'Pendiente por leer';

export interface Libro {
  _id?: string;

  titulo: string;
  autor: string;
  anio: number;
  totalPaginas: number;

  genero: string;
  portada: string;

  estadoLectura: EstadoLectura;
  tiempoLectura?: string;

  disponible: boolean;
  prestadoA?: string;
  fechaPrestamo?: string;

  favorito: boolean;
  calificacion: number;

  resena: string;
}

File: Biblioteca/backend/app/models/libro_model.py
from pydantic import BaseModel

class Libro(BaseModel):
    titulo: str
    autor: str
    anio: int
    totalPaginas: int

    genero: str

    portada: str | None = None

    estadoLectura: str
    tiempoLectura: str | None = None

    disponible: bool

    prestadoA: str | None = None
    fechaPrestamo: str | None = None

    favorito: bool
    calificacion: int

    resena: str

    usuarioId: str | None = None